### PR TITLE
Better C# build instructions.

### PR DESCRIPTION
**Pull request by jtattermusch**
This is to fix some of the problems mentioned in https://github.com/grpc/grpc/issues/15100.

Supersedes https://github.com/grpc/grpc/pull/15099.

### CODE REVIEW

**Comment by AtosNicoS** [comment]
Please also add the information about the dependencies. You dont have to specify the exact name of the package for each distribution, but it should be listed at least to know what is missing. For example I spent a lot of time searching that some dependencies are missing. Finding the correct name for ArchLinux was easy though.

* python2
* python2-six 
* python2-google-api-python-client
* dotnet-sdk
* mono

Also please change `python` to `python2`. Or change the shebang in the .py script, make it executeable and remove it completely.

File: src/csharp/README.md [hunk 1]
```diff
@@ -39,10 +39,14 @@ If you are a user of gRPC C#, go to Usage section above.
 
 **Windows, Linux or Mac OS X**
 
-- The easiest way to build is using the `run_tests.py` script that will take care of building the `grpc_csharp_ext` native library:
+- The easiest way to build is using the `run_tests.py` script that will take care of building the `grpc_csharp_ext` native library.
+  You might first need to install the prerequisites mentioned in [INSTALL.md](../../INSTALL.md#pre-requisites).
   ```
+  # NOTE: make sure all necessary git submodules with dependencies
```
Comment (apolcyn) [inline src/csharp/README.md]: nit: these "```" ticks are showing up in the rendered md file, there might need to be a newline on top

**Comment by jtattermusch** [comment]
@AtosNicoS   I mentioned dotnet SDK and mono explicitly in the  instructions.

The python2 and python2-six I haven't mentioned as that would require changes across the whole repository (run_tests.py is used by all languages), so this PR is not a good place to change this.

I haven't mentioned python2-google-api-python-client because that one is not necessary (after https://github.com/grpc/grpc/pull/15180 being in).

**Comment by AtosNicoS** [comment]
Okay, I overlooked the mono and dotnet-sdk instructions, sorry for that.

The python2 error is quite simple to fix for a developer, so thats fine. I'd still recommend to change the shebang to python2 and remove the `python` command from all readmes. If you ever plan to switch to python3, then its already fixed.
https://github.com/grpc/grpc/blob/5947d337fd94d156ba937555fb5492df24d7f3ba/tools/run_tests/run_tests.py#L1

Another thing I'd add is a small hint where the compiled binaries are located and how to use them.

Thanks for clarifying the build instructions.

**Comment by jtattermusch** [comment]
@AtosNicoS   we might look into the python version problem separately (a more major overhaul is probably needed)

**Pull request merged**